mr/mr.cc

```diff
@@ -96,25 +96,22 @@ struct ShardVisitor {
 
 }  // namespace
 
+void AnyFreqMap::CheckType(const std::type_info& t) const {
+  CHECK(t == any_.type()) << "Mismatch between " << t.name() << " and " << any_.type().name();
+}
+
 RawContext::RawContext() { metric_map_.set_empty_key(StringPiece{}); }
 
 RawContext::~RawContext() {}
 
-FrequencyMap<uint32_t>& RawContext::GetFreqMapStatistic(const std::string& map_id) {
-  auto res = freq_maps_.emplace(map_id, nullptr);
-  if (res.second) {
-    res.first->second.reset(new FrequencyMap<uint32_t>);
-  }
-  return *res.first->second;
-}
-
-const FrequencyMap<uint32_t>* RawContext::FindMaterializedFreqMapStatistic(
-    const std::string& map_id) const {
+const AnyFreqMap *
+RawContext::FindMaterializedFreqMapStatisticNotNull(const std::string& map_id) const {
```

Comment:
why did you change the name? it does return nullptr

Reply:
Because it has CHECK_NOTNULL in the beginning. The NotNull is for the input, not the output.
Weil sie CHECK_NOTNULL am Anfang hat. Die NotNull ist für die Eingabe, nicht die Ausgabe.

Reply:
I find it confusing. I would expect `void* FooBarNotNull` not to return null.